server/routerlicious/packages/routerlicious-base/src/riddler/tenantManager.ts

```diff
@@ -455,4 +558,44 @@ export class TenantManager {
         const accessInfo = JSON.parse(this.secretManager.decryptSecret(encryptedAccessInfo));
         return accessInfo;
     }
+
+    private decryptCachedKeys(cachedKey: string) {
+        const keys = JSON.parse(cachedKey);
+        return keys.key2 === "" ? {
+                key1: this.secretManager.decryptSecret(keys.key1),
+                key2: "",
+            } : {
+                key1: this.secretManager.decryptSecret(keys.key1),
+                key2: this.secretManager.decryptSecret(keys.key2),
+            };
+    }
+
+    private async getCacheKey(tenantId: string): Promise<string> {
+        return this.cache?.get(`tenantKeys:${tenantId}`);
+    }
+
+    private async deleteCacheKey(tenantId: string): Promise<boolean> {
+        return this.cache?.delete(`tenantKeys:${tenantId}`);
+    }
+
+    private async setCacheKey(tenantId: string, value: ITenantKeys, metric?: Lumber<LumberEventName.RiddlerFetchTenantKey>): Promise<void> {
```

Comment:
Regarding the parameter `Lumber<LumberEventName.RiddlerFetchTenantKey>`. We call `setCacheKey()` in other places other than in the `getTenantKeys()` method. It is true that `getTenantKeys()` is the only place where we pass a metric to `setCacheKey()` so far, but in case we want to allow the other callers to pass other metrics as well, having this parameter be specific to `LumberEventName.RiddlerFetchTenantKey` might not be the best idea. Perhaps we can update this to be `Lumber<LumberEventName>` instead? (or even `Lumber<string>` if the other suggestion doesn't work)

Reply:
Changing this to `Lumber` since `Lumber<LumberEventName>` is the `Lumber` default value